crates/compiler/can/src/module.rs

```diff
@@ -789,16 +829,38 @@ fn fix_values_captured_in_closure_defs(
         );
     }
 
-    // TODO mutually recursive functions should both capture the union of both their capture sets
+    for def in defs.iter_mut() {
+        fix_values_captured_in_closure_def(def, no_capture_symbols, closure_captures);
+    }
 
+    // Mutually recursive functions should both capture the union of all their capture sets
+    //
+    // Really unfortunate we make a lot of clones here, can this be done more efficiently?
```

Comment:
I only see a clone of the total capture set below. What am I missing?

Reply:
We make a clone of the capture set for each lambda in a module. We don't remove them when they leave their scopes. Maybe that's fine, since it gets wiped away immediately after this fixing pass

Reply:
the `defs` here is just the definitions that are mutually recursive. Also, mutually recursive function definitions that are not at the top level, that must be really rare